src/main/java/fi/jubic/quanta/controller/ExternalClientController.java

```diff
@@ -17,29 +21,73 @@
 
     private final ExternalClientDao externalClientDao;
     private final TaskDao taskDao;
+    private final UserDao userDao;
 
     @Inject
     ExternalClientController(
             ExternalClientDomain externalClientDomain,
             ExternalClientDao externalClientDao,
-            TaskDao taskDao
+            TaskDao taskDao,
+            UserDao userDao
     ) {
         this.externalClientDomain = externalClientDomain;
         this.externalClientDao = externalClientDao;
         this.taskDao = taskDao;
+        this.userDao = userDao;
     }
 
-    public ExternalClient generateExternalClient(Long taskId, ExternalClient externalClient) {
+    public List<ExternalClient> search(ExternalClientQuery query) {
+        return externalClientDao.search(query);
+    }
+
+    public ExternalClient generateExternalClient(ExternalClientView externalClientView) {

```

Comment:
Using `ExternalClientView` here adds a lot of noise. Models should be used in the controller methods as often as possible, but here we would need yet another request model.

I think the methods here should be

```java
public ExternalClient createExternalClient(String name, String description) [...]

public ExternalClient createExternalClient(String name, String description, Long taskId) [...]
```

Reply:
Changed request models